src/utils/api.ts

```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+
+export const api = async <Response>(args: {
+  url: string;
+  params?: Record<string, string | number>;
+}) => {
+  try {
+    const result = await axios({
```

Comment:
в axios можно прокинуть типы, тогда дальнейшая логика с `as` не понадобится:
```suggestion
    const result = await axios<Response>({
```